server/routes/api/notifications/notifications.ts

```diff
@@ -49,4 +56,84 @@ router.post(
   handleUnsubscribe
 );
 
+router.post(
+  "notifications.list",
+  auth(),
+  pagination(),
+  validate(T.NotificationsListSchema),
+  transaction(),
+  async (ctx: APIContext<T.NotificationsListReq>) => {
+    const eventType = ctx.input.body.eventType ?? ctx.input.query.eventType;
+    const archived = ctx.input.body.archived ?? ctx.input.query.archived;
+    const user = ctx.state.auth.user;
+    let where: WhereOptions<Notification> = {
+      userId: user.id,
+    };
+    if (eventType) {
+      where = { ...where, event: eventType };
+    }
+    if (archived) {
+      where = {
+        ...where,
+        archivedAt: {
+          [Op.ne]: null,
+        },
+      };
+    }
+    const [notifications, unseenCount] = await Promise.all([
+      Notification.scope(["withUser", "withActor"]).findAll({
+        where,
+        order: [["createdAt", "DESC"]],
+        offset: ctx.state.pagination.offset,
+        limit: ctx.state.pagination.limit,
+      }),
+      Notification.count({
+        where: {
+          ...where,
+          viewedAt: {
+            [Op.is]: null,
+          },
```

Comment:
Note that there's a subtle assumption here. The count actually changes based on other `where` attributes. For example, if `archived` is received as `true`, then this count consists of notifications which are both archived _and_ unseen. On the contrary, the intended purpose may be to return the count of _all_ unseen notifications without extra filters. Let me know your thoughts @tommoor